x-pack/docs/en/security/backup-security-configuration.asciidoc

```diff
@@ -0,0 +1,263 @@
+
+[float]
+[[backup-security-configuration-inside-cluster]]
+==== Security configuration inside the cluster
+
+The API calls act on configuration information that is stored in the global
+cluster metadata as well as inside a dedicated {es} index. This index is named
+`.security-6` in the 6.x versions and `.security-7` in the 7.x releases. The
+`.security` alias always points to the appropriate index. This index contains
+the data which is *not* part of an ordinary configuration backup. This data
+describes:
+
+* the definition of users in the native realm (including hashed passwords)
+* role definitions
+* role mappings
+* application privileges
+* API keys
+
+Be advised that the index format is only compatible over a single major version.
+
+TIP: The `elasticsearch.yml` and `elasticsearch.keystore` files and the global
+cluster metadata contain settings for all the {stack} features. The
+`.security` index and all the other
+<<security-files,security configuration files>> contain configuration data
+exclusively for the {security-features}. It is easiest to backup and restore all
+the configuration information as a whole, rather than trying to cherry-pick only
+the security details.
+
+[float]
+[[backup-security-files]]
+==== Backing up configuration files
+
+The simplest way to backup configuration files is to copy the `$ES_PATH_CONF`
+directory. Overall, this procedure is identical to a conventional backup.
+
+NOTE: In a typical {es} deployment, all the configuration files are identical on
+each node.
+
+It is recommended that you back up the configuration files every time they are
+modified. This is not as difficult as it sounds because, in a typical
+deployment, after you achieve a desired configuration, you update these files
+infrequently. Configuration files purposefully contain only the settings that
+change rarely or are required for cluster formation or require file system
+privileges.
+
+TIP: For the YAML text files, you also have the option of storing diffs against a
+baseline or checking them into `git`. This method makes it easier for the {es}
+administrator to review the changes.
+
+The binary and encoded files, such as the private keys and the secure store
+file, are extra sensitive; they require restricted read access.
+
+[float]
+[[backup-security-snapshots]]
+==== Snapshotting the security index and the global cluster metadata
+
+To back up API-based configuration, which is stored inside the cluster, you use
+the <<modules-snapshots,regular snapshot functionality>>. The snapshot API
+saves the settings, the type mapping, and the contents of any index to a
+location on a cloud storage service or on a shared network file system. This is
+used to snapshot the `.security` index. The snapshot operation can also
+optionally capture the persistent settings from the global cluster metadata.
+
+. Create a repository that you can use to backup the `.security` index.
+Ideally this repository is dedicated only to this scope.
```

Comment:
I think if we're going to say this (`Ideally ...`) then we need to explain why.

Personally, I would recommend that customers think about snapshoting the `.kibana` data to this same repo. 

If I were the admin, I would probably have:
- One repo for "my" data. It would have have the indices, and we'd have to decide whether the costs are justified depending on data volumes (do we really need to keep snapshots of 180 days worth of log files)?
- Another repo for "system" data. `.security`, `.kibana*`, `.ml-*` and cluster state

That might be a bad idea (please say so), if we're going to same a particular configuration is "ideal" we need to have reasons why.

Reply:
"system" data and cluster state inside a separate repo sounds good to me. Probably `.kibana` contains even more sensitive data (un-hashed credentials), but I don't believe that there is any compounded risk if `.security` and `.kibana` get together.

However, for this iteration, I would keep `.kibana` and `.ml-*` out of the discussion, for the reason that we might make assertions that need testing so they don't become stale. If you feel we should mention them in this context only, I am fine with that.

Reply:
I'm happy to be a little big vague, and just talk about separating system indices into their own repository. And if ES is their primary datastore, their own apps might have a "configuration" index that they decide to snapshot alongside the `.` indices. That's up to them.

I know that our support team often encourage people to take snapshots of the kibana indices, even if they can't justify the cost of snapshotting their own data. It would be good if our instructions here aligned with the advice they give.

When we've got a finished version, it's probably worth asking someone from `@elastic/support-tech-leads`  to review so they can update their internal docs, or tell us if there's conflicting advice.